app/models/user.rb

```diff
@@ -3,8 +3,51 @@ class User < ActiveRecord::Base
   # validates password and password_confirmation; no validates_presence_of necessary!!!
   has_secure_password validations: true
   has_many :queue_items, -> { order('order_value') }
+  has_many :reviews
 
   validates_presence_of :email, :full_name
   validates_uniqueness_of :email
 
+  def update_queue_items(queue_items_hash)
+    ActiveRecord::Base.transaction do
+      queue_items_hash.each do |queue_item_element|
+        queue_item = queue_items.find_by(id: queue_item_element['id'])
+        next if queue_item.nil?
+        update_order_value queue_item, queue_item_element['order_value']
```

Comment:
`update_order_value` + `update_rating`= a lot of complex/complicated code that can be actually refactored by using mass assignment and `update!`. Check the solution video and see if you like it better.